util/deephash/deephash.go

```diff
@@ -336,3 +344,43 @@ func (h *hasher) hashMapFallback(v reflect.Value) (acyclic bool) {
 	w.WriteString("}\n")
 	return acyclic
 }
+
+func (h *hasher) hashType(t reflect.Type) {
+	// Unwrap anonymous pointers to named types since they are so common.
+	name := t.Name()
+	isAnonPtr := name == "" && t.Kind() == reflect.Ptr
+	if isAnonPtr {
+		t = t.Elem()
+		name = t.Name()
+	}
+
+	// Format the qualified name of the type.
+	if name != "" {
+		size := h.scratch[:8]
+		qualifiedName := h.scratch[8:8] // reserve space for size
+		if isAnonPtr {
+			qualifiedName = append(qualifiedName, '*')
+		}
+		if pkgPath := t.PkgPath(); pkgPath != "" {
+			qualifiedName = strconv.AppendQuote(qualifiedName, pkgPath)
+		}
+		qualifiedName = append(qualifiedName, '.')
+		qualifiedName = append(qualifiedName, name...)
+		binary.LittleEndian.PutUint64(size, uint64(len(qualifiedName)))
+		h.bw.Write(size)
```

Comment:
It probably doesn't matter for performance, but you could do this in a single write with a bit of refactoring above.

Reply:
Got rid of this approach.